src/app/models/state.model.ts

```diff
@@ -0,0 +1,13 @@
+import {Recipe} from './recipe.model';
+import {Filters} from './filters.model';
+
+export interface AppState { recipes: { [id: string]: Recipe }, list: string[], filters: Filters };
+export interface State { app: AppState }; // this will also contain router state
+
+export const initialState: State = {
```

Comment:
I think categories, images, and other relationship data should be introduced after the angular2-jsonapi integration, instead of expanding out my already hacky api service / normalizer.